src/librustdoc/json/mod.rs

```diff
@@ -1,47 +1,238 @@
+        Ok((
+            JsonRenderer {
+                index: Rc::new(RefCell::new(FxHashMap::default())),
+                out_path: options.output,
+            },
+            krate,
+        ))
     }
 
-    fn item(&mut self, _item: clean::Item, _cache: &Cache) -> Result<(), Error> {
-        unimplemented!()
+    /// Inserts an item into the index. This should be used rather than directly calling insert on
+    /// the hashmap because certain items (traits and types) need to have their mappings for trait
+    /// implementations filled out before they're inserted.
+    fn item(&mut self, item: clean::Item, cache: &Cache) -> Result<(), Error> {
+        // Flatten items that recursively store other items
+        item.inner.inner_items().for_each(|i| self.item(i.clone(), cache).unwrap());
+
+        let id = item.def_id;
+        let mut new_item: types::Item = item.into();
+        if let types::ItemEnum::TraitItem(ref mut t) = new_item.inner {
+            t.implementors = self.get_trait_implementors(id, cache)
+        } else if let types::ItemEnum::StructItem(ref mut s) = new_item.inner {
+            s.impls = self.get_impls(id, cache)
+        } else if let types::ItemEnum::EnumItem(ref mut e) = new_item.inner {
+            e.impls = self.get_impls(id, cache)
+        }
+
+        self.index.borrow_mut().insert(id.into(), new_item);
+        Ok(())
     }
 
     fn mod_item_in(
         &mut self,
-        _item: &clean::Item,
-        _item_name: &str,
-        _cache: &Cache,
+        item: &clean::Item,
+        _module_name: &str,
+        cache: &Cache,
     ) -> Result<(), Error> {
-        unimplemented!()
+        use clean::types::ItemEnum::*;
+        if let ModuleItem(m) = &item.inner {
+            for item in &m.items {
+                match &item.inner {
+                    // These don't have names so they don't get added to the output by default
+                    ImportItem(_) => self.item(item.clone(), cache).unwrap(),
+                    ExternCrateItem(_, _) => self.item(item.clone(), cache).unwrap(),
+                    ImplItem(i) => {
+                        i.items.iter().for_each(|i| self.item(i.clone(), cache).unwrap())
+                    }
+                    _ => {}
+                }
+            }
+        }
+        self.item(item.clone(), cache).unwrap();
+        Ok(())
     }
 
     fn mod_item_out(&mut self, _item_name: &str) -> Result<(), Error> {
-        unimplemented!()
+        Ok(())
     }
 
-    fn after_krate(&mut self, _krate: &clean::Crate, _cache: &Cache) -> Result<(), Error> {
-        unimplemented!()
+    fn after_krate(&mut self, krate: &clean::Crate, cache: &Cache) -> Result<(), Error> {
+        debug!("Done with crate");
+        let mut index = (*self.index).clone().into_inner();
+        index.extend(self.get_trait_items(cache));
+        let output = types::Crate {
+            root: types::Id(String::from("0:0")),
+            version: krate.version.clone(),
```

Comment:
This is empty if the version is unknown. Should it instead be `null`?

Reply:
Also, it might be better to name this `crate_version` to avoid confusing with `format_version`.

Reply:
> This is empty if the version is unknown. Should it instead be null?

Oops, this was a bug in jq - it prints `null` as black on black, making it invisible.